x-pack/platform/plugins/shared/ml/server/routes/data_frame_analytics.ts

```diff
@@ -470,9 +471,25 @@ export function dataFrameAnalyticsRoutes(
                 // If user does have privilege to delete the index, then delete the index
                 if (userCanDeleteDestIndex) {
                   try {
-                    await client.asCurrentUser.indices.delete({
+                    // It's possible that the destination index has been reindexed with a new name
+                    // so if it's an alias first, then delete the real index
+                    let destinationIndexToDelete = destinationIndex;
+                    const alias = await client.asCurrentUser.indices.getAlias({
```

Comment:
Will this 404 if the alias doesn't exist and then not continue with the delete?

Reply:
The `getAlias` always return with a response here 
```
{ dest_glass_010: { aliases: {} } } // note the aliases object is just empty
```` 
 so the index will still be deleted

Reply:
Ok cool. Looking at the documentation it'll error if the index doesn't exist.
Which I guess is ok as far as needing to delete the index, but I wonder if we should then delete this orphaned alias?

Reply:
Yeah it's a bit strange why the behavior of getAlias is different, but just for clarity, I added an ignore 404 for extra clarity [`1d644d8` (#213306)](https://github.com/elastic/kibana/pull/213306/commits/1d644d8c89759f814bad55c110838181293e3d92)

Deleting the target index/destination index already deletes the alias so I don't think we need that extra step of deleting the alias.